providers/azure.go

```diff
@@ -151,6 +157,60 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (s
 		RefreshToken: jsonResponse.RefreshToken,
 	}
 	return
+
+}
+
+// RefreshSessionIfNeeded checks if the session has expired and uses the
+// RefreshToken to fetch a new ID token if required
+func (p *AzureProvider) RefreshSessionIfNeeded(ctx context.Context, s *sessions.SessionState) (bool, error) {
+	if s == nil || s.ExpiresOn.After(time.Now()) || s.RefreshToken == "" {
+		return false, nil
+	}
+
+	origExpiration := s.ExpiresOn
+
+	err := p.redeemRefreshToken(ctx, s)
+
+	if err != nil {
```

Comment:
Nit: Please pair error declaration and checks, this is how they normally are in go 
```suggestion
	err := p.redeemRefreshToken(ctx, s)
	if err != nil {
```